kart/tabular/export.py

```diff
@@ -0,0 +1,304 @@
+    "table-export",
+    cls=KartCommand,
+    hidden=True,
+    context_settings=dict(ignore_unknown_options=True),
+)
+@click.pass_context
+@call_and_exit_flag(
+    "--list-formats",
+    callback=list_export_formats,
+    help="List available export formats, and then exit",
+)
+@click.option(
+    "--ref", default="HEAD", help="The revision at which to export the dataset."
+)
+@click.option(
+    "--layer",
+    help="Name of the layer to create inside the given destination. Defaults to the dataset path.",
+)
+@click.option(
+    "--layer-creation-option",
+    "-lco",
+    multiple=True,
+    help="A layer creation option (-lco) flag forwarded to the GDAL OGR driver. Example: -lco VERSION=1.0",
+)
+@click.option(
+    "--dataset-creation-option",
+    "-dsco",
+    multiple=True,
+    help="A dataset creation option (-dsco) flag forwarded to the GDAL OGR driver. Example: -dsco LAUNDER=YES",
+)
+@click.option(
+    "--primary-key-as-field/--no-primary-key-as-field",
+    is_flag=True,
+    default=True,
+    help="Include the primary key column as a regular field that is sent to the GDAL driver.",
+)
+@click.option(
+    "--primary-key-as-fid/--no-primary-key-as-fid",
+    is_flag=True,
+    default=False,
+    help="Report to the GDAL driver that the FID of each feature is the primary key value.",
+)
+@click.option(
+    "--override-geometry",
+    help="Override the geometry type to something more specific than in the Kart dataset.",
+)
+@click.option(
+    "--drop-null-geometry-features",
+    is_flag=True,
+    help="Skips export of those features where the geometry is null.",
+)
+@click.argument(
+    "args",
+    nargs=-1,
+    metavar="DATASET [EXPORT_TYPE:]DESTINATION",
+    shell_complete=repo_path_completer,
+)
+def table_export(
+    ctx,
+    ref,
+    layer,
+    layer_creation_option,
+    dataset_creation_option,
+    primary_key_as_field,
+    primary_key_as_fid,
+    override_geometry,
+    drop_null_geometry_features,
+    args,
+    **kwargs,
+):
+    """
+    Experimental export command - exports a tabular kart dataset at a particular commit.
+
+    Uses GDAL's OGR drivers to do so - consult https://gdal.org/drivers/vector/index.html
+    to find the options specific to a particular driver, which can then be set using
+    -dsco and -lco just as when using `ogr2ogr`.
+    """
+    if len(args) < 2:
+        raise click.UsageError(
+            "Usage: kart table-export DATASET [EXPORT_TYPE:]DESTINATION"
+        )
+
+    repo = ctx.obj.repo
+
+    ds_path = args[0]
+    destination_spec = args[1]
+
+    try:
+        dataset = repo.datasets(ref)[ds_path]
+    except KeyError:
+        raise click.UsageError(f"No such dataset: {ds_path}")
+
+    if dataset.DATASET_TYPE != "table":
+        raise click.UsageError(f"Dataset {ds_path} is not a vector or tabular dataset")
+
+    driver, destination = get_driver(destination_spec)
+    out_ds = driver.CreateDataSource(destination, options=list(dataset_creation_option))
```

Comment:
I'm unclear what this would do if the output already exists? I guess to allow multiple datasets to end up in the same FGDB/GPKG/etc then we should generally allow this.

For some formats the output is a directory (shapefiles); for others it's a file (GPKG), which complicates things a bit.

We don't want the working copy overwritten, that would be bad.